build-and-stage.ps1

```diff
@@ -126,6 +126,22 @@ else
     Pop-Location
 }
 
+Write-Host ""
+Write-Host "============================================================"
+Write-Host "Pre-build step"
+Write-Host "============================================================"
+$preBuildScript = "custom-steps/$packageNameOnly/pre-build.ps1"
+if (Test-Path -Path $preBuildScript -PathType Leaf) {
+    Write-Host "Executing: $preBuildScript..."
+    $invokePrefix = ""
+    if(-not $IsOnWindowsOS) {
+      $invokePrefix = "pwsh "
+    }
+    Invoke-Expression "$invokePrefix./$preBuildScript"
```

Comment:
What does the extra `.` after `$invokePrefix.` mean?

Reply:
Oh nevermind, I get it. There's a `./` to execute the script.

Would it be more readable to add a utility function `Invoke-Powershell` that figures out the right prefix and then does the invocation?

I was going to suggest eliminating some duplicate with this part and the postBuildScript.

Reply:
Replaced this with Invoke-PowerShell: 
- https://github.com/TechSmith/ThirdParty-Packages-vcpkg/pull/1/files#diff-a30b3ca7930c449f10475692234c9695f64e3b25fc9233d11167e18f03b6f2dcR185-R186
- https://github.com/TechSmith/ThirdParty-Packages-vcpkg/pull/1/files#diff-a30b3ca7930c449f10475692234c9695f64e3b25fc9233d11167e18f03b6f2dcR145
- https://github.com/TechSmith/ThirdParty-Packages-vcpkg/pull/1/files#diff-e5e31509d9c00785953ee352878be71b8843aae1f92c12c6cb4fbc0b210c1b1fR80-R102